qiskit/circuit/qpy_serialization.py

```diff
@@ -10,1913 +10,17 @@
-                        if start > len(circ.qubits):
-                            bits = [bit_type() for _ in range(start - bit_len)]
-                            circ.add_bits(bit_len)
-                        if in_circuit:
-                            circ.add_register(reg)
-                        out_registers[bit_type_label][register_name] = reg
-                else:
-                    for index in indices:
-                        if bit_type_label == "q":
-                            bit_len = len(circ.qubits)
-                        else:
-                            bit_len = len(circ.clbits)
-                        # Add any missing bits
-                        bits = [bit_type() for _ in range(index + 1 - bit_len)]
-                        circ.add_bits(bits)
-                        if index in register_bits:
-                            raise QiskitError("Duplicate register bits found")
-                        register_bits.add(index)
-                    if bit_type_label == "q":
-                        bits = [circ.qubits[i] for i in indices]
-                    else:
-                        bits = [circ.clbits[i] for i in indices]
-                    reg = reg_type(name=register_name, bits=bits)
-                    if in_circuit:
-                        circ.add_register(reg)
-                    out_registers[bit_type_label][register_name] = reg
-        # If we don't have sufficient bits in the circuit after adding
-        # all the registers add more bits to fill the circuit
-        if len(circ.qubits) < num_qubits:
-            qubits = [Qubit() for _ in range(num_qubits - len(circ.qubits))]
-            circ.add_bits(qubits)
-        if len(circ.clbits) < num_clbits:
-            clbits = [Clbit() for _ in range(num_qubits - len(circ.clbits))]
-            circ.add_bits(clbits)
-    else:
-        circ = QuantumCircuit(
-            num_qubits,
-            num_clbits,
-            name=name,
-            global_phase=global_phase,
-            metadata=metadata,
-        )
-    custom_instructions = _read_custom_instructions(file_obj, version, vectors)
-    for _instruction in range(num_instructions):
-        _read_instruction(file_obj, circ, out_registers, custom_instructions, version, vectors)
-    for vec_name, (vector, initialized_params) in vectors.items():
-        if len(initialized_params) != len(vector):
-            warnings.warn(
-                f"The ParameterVector: '{vec_name}' is not fully identical to its "
-                "pre-serialization state. Elements "
-                f"{', '.join([str(x) for x in set(range(len(vector))) - initialized_params])} "
-                "in the ParameterVector will be not equal to the pre-serialized ParameterVector "
-                f"as they weren't used in the circuit: {circ.name}",
-                UserWarning,
-            )
 
-    return circ
+from qiskit.qpy import dump, load
```

Comment:
I didn't realize this (probably they should not directly call private anyways). Sounds like this is where your neko comes into play 🐱

Reply:
I agree, it's not something they should be doing. But while they are we can be kind and just re-export it :)  (I also think the ibm provider should have it's own qpy generation module to avoid https://github.com/Qiskit/qiskit-ibm-runtime/issues/147 )

Yeah, once we have the jobs set up with qiskit-neko in place it could test this type of thing. The bigger issue with the ibm provider packages is we probably still can't run those in ci because of the execution time and credentials needs